vault/acl_test.go

```diff
@@ -801,26 +801,29 @@ func TestACL_CreationRace(t *testing.T) {
 		t.Fatalf("err: %v", err)
 	}
 
-	var wg sync.WaitGroup
 	stopTime := time.Now().Add(20 * time.Second)
 
 	for i := 0; i < 50; i++ {
-		wg.Add(1)
+		errCh := make(chan error)
+
 		go func() {
-			defer wg.Done()
 			for {
 				if time.Now().After(stopTime) {
+					errCh <- nil
 					return
 				}
 				_, err := NewACL(namespace.RootContext(nil), []*Policy{policy})
 				if err != nil {
-					t.Fatalf("err: %v", err)
+					errCh <- fmt.Errorf("err: %v", err)
+					return
 				}
 			}
 		}()
+		err = <-errCh
```

Comment:
This would block the for loop until the error is received before the next goroutine gets called.